### DISCUSSION TITLE
Breaking Change: Removal of curl from .NET 5.0 runtime Docker images

### DISCUSSION THREAD

**Discussion by mthalman** [dotnet/dotnet-docker#3678] [Announcements]
# Removal of curl from runtime Docker images

The `curl` package has been removed from the `runtime` and `aspnet` Linux images with the release of .NET 5.0. This is considered a breaking change which may impact your build or application when upgrading to .NET 5.0.

## Details

Our primary goal in producing Docker images for .NET is to reduce the occurrences of security vulnerabilities. We determined that the use of `curl` in these images wasn't a common enough scenario to justify the increased security surface area it added. If you rely on `curl` as part of `docker build`, you are encouraged to install it yourself in your Dockerfile, or rely on multi-stage build, by using a buildpack-deps stage, as we do for [.NET runtime images](https://github.com/dotnet/dotnet-docker/blob/master/src/runtime/5.0/focal/amd64/Dockerfile#L5). If you only require `curl` when consuming the `sdk` image, you won't be affected since `curl` is included in that image.

### Related Issues

* [Remove curl from runtime image](https://github.com/dotnet/dotnet-docker/issues/1645)

**Comment by HaroonSaid** [comment]
noticed that wget is also not present, whats the recommend path for doing container health check?

**Comment by MichaelSimons** [comment]
> noticed that wget is also not present, whats the recommend path for doing container health check?

A couple options that come to mind.
1. Use an Alpine based image because it includes wget.
2. Add your tool of choice in your derived image.

**Reply by aa-ko** [reply]
Sorry for necroposting, but I gotta ask if there is any recommended solution when working with `dotnet publish` to create container images.

As far as I can tell, all dotnet images have had `curl` removed, so I guess I'd have to build a custom base image like this:
- Use `dotnet-runtime` as a base
- Install `curl`
- Publish image internally and use as base for my production build

Is there an easier way to do this if I want to keep using `dotnet publish`?

**Reply by baronfel** [reply]
There's nothing easier today, no - we're tracking this need at https://github.com/dotnet/sdk-container-builds/issues/125, but it needs more design work, especially around how to do this in a way that doesn't completely destroy MSBuild incrementality.

**Comment by chrisbecke** [comment]
Might want to update [your docs ](https://docs.microsoft.com/en-us/visualstudio/containers/container-tools-react?view=vs-2019) then.
